base/bucket_gocb.go

```diff
@@ -104,7 +105,18 @@ func GetCouchbaseBucketGoCB(spec BucketSpec) (bucket *CouchbaseBucketGoCB, err e
 		EnableGoCBLogging()
 	}
 
-	cluster, err := gocb.Connect(spec.Server)
+	connSpec, err := gocbconnstr.Parse(spec.Server)
+	if err != nil {
+		return nil, err
+	}
+
+	// Increase the number of idle connections per-host to fix SG #3534
+	if connSpec.Options == nil {
+		connSpec.Options = map[string][]string{}
+	}
+	connSpec.Options["http_max_idle_conns_per_host"] = []string{"100"}
```

Comment:
Can you stick http_max_idle_conns_per_host in a constant?  The x.509 work is going to add three more connSpec options - I'll stick those in the same place.

Reply:
Fixed